### ISSUE TITLE
allow(ptr_arg) in argument position

### ISSUE THREAD

**Issue by schatzmeister** [rust-lang/rust-clippy#5644]
Clippy messages for ptr_arg aren’t disabled when the allow attribute is directly on the argument:

```rust
fn foo(
    // Clippy still warns here:
    #[allow(clippy::ptr_arg)] 
    arg: &Vec<u16>,
) {
    // ...
}
```

Putting it to the top of the function instead works fine:

```rust
// Everything works
#[allow(clippy::ptr_arg)] 
fn foo(arg: &Vec<u16>) {
    // ...
}
```

I think this should be allowed, since it works with other lints.

Link to playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b33ea40fad2a66997225dd21f858230a
